libraries/common/src/vrf_wrapper.cpp

```diff
@@ -29,12 +29,17 @@ std::optional<vrf_proof_t> getVrfProof(vrf_sk_t const &sk, bytes const &msg) {
   return {};
 }
 
-std::optional<vrf_output_t> getVrfOutput(vrf_pk_t const &pk, vrf_proof_t const &proof, bytes const &msg) {
+std::optional<vrf_output_t> getVrfOutput(vrf_pk_t const &pk, vrf_proof_t const &proof, bytes const &msg, bool strict) {
   vrf_output_t output;
-  // crypto_vrf_verify return 0 on success!
-  if (!crypto_vrf_verify(output.data(), const_cast<unsigned char *>(pk.data()),
-                         const_cast<unsigned char *>(proof.data()), const_cast<unsigned char *>(msg.data()),
-                         msg.size())) {
+  if (strict) {
+    // crypto_vrf_verify return 0 on success!
+    if (!crypto_vrf_verify(output.data(), const_cast<unsigned char *>(pk.data()),
+                           const_cast<unsigned char *>(proof.data()), const_cast<unsigned char *>(msg.data()),
+                           msg.size())) {
+      return output;
+    }
+  } else {
+    crypto_vrf_proof_to_hash(output.data(), const_cast<unsigned char *>(proof.data()));
```

Comment:
What is this function doing?

Reply:
I did the measurement and verify vrf together with getting the output takes about 300microseconds, just getting the output without full validation takes only 20 microseconds. If there is 50 votes per block for 5m blocks this is about 20 hours of processing to verify all the votes for each block. This is by far the biggest CPU bottleneck of current syncing/rebuilding.